src/utils/mtev_watchdog.c

```diff
@@ -114,45 +131,56 @@ void mtev_watchdog_ratelimit(int retry_val, int span_val) {
 
 /* Watchdog stuff */
 static pid_t watcher = -1;
-static int *lifeline = NULL;
-static unsigned long last_tick_time() {
-  static struct timeval lastchange = { 0, 0 };
-  static int lastcheck = 0;
-  struct timeval now, diff;
+static mtev_watchdog_t *mmap_lifelines = NULL;
+static unsigned long last_tick_time(mtev_watchdog_t *lifeline, struct timeval *now) {
+  struct timeval diff;
 
-  mtev_gettimeofday(&now, NULL);
-  if(lastcheck != *lifeline) {
-    lastcheck = *lifeline;
-    memcpy(&lastchange, &now, sizeof(lastchange));
+  if(lifeline == NULL) lifeline = mmap_lifelines;
+
+  if(lifeline->parent_view.last_ticker != lifeline->ticker) {
+    lifeline->parent_view.last_ticker = lifeline->ticker;
+    memcpy(&lifeline->parent_view.last_changed, now, sizeof(*now));
   }
-  if(lastchange.tv_sec == 0) return 0;
+  if(lifeline->parent_view.last_changed.tv_sec == 0) return 0;
 
-  sub_timeval(now, lastchange, &diff);
+  sub_timeval(*now, lifeline->parent_view.last_changed, &diff);
   return (unsigned long)diff.tv_sec;
 }
-static void it_ticks_crash() {
-  if(lifeline) lifeline[1] = CRASHY_CRASH;
+static void it_ticks_crash(mtev_watchdog_t *lifeline) {
+  if(lifeline == NULL) lifeline = mmap_lifelines;
```

Comment:
possibly this will be explained by a forthcoming PR, but since I'm only looking at this PR, why not remove all the
```
  if(lifeline == NULL) lifeline = mmap_lifelines;
```
statements, and just call these `it_ticks_*` functions with `mmap_lifelines`?